crates/node/src/shims/abcipp_shim.rs

```diff
@@ -177,6 +191,62 @@ impl AbcippShim {
             }
         }
     }
+
+    fn update_snapshot_task(&mut self, take_snapshot: TakeSnapshot) {
+        if self
+            .snapshot_task
+            .as_ref()
+            .map(|t| t.is_finished())
+            .unwrap_or_default()
+        {
```

Comment:
nit: I think this code is easier to read if we bind this logic to a variable, and invert the condition to do an early return
```rs
if !snapshot_taken {
    return;
}
```

Reply:
I'm confused what you mean. Right now, we don't early return if there is current snapshot task or not.

Reply:
if I understand correctly, a snapshot is being taken when `self.snapshot_task.is_some()` and `!t.is_finished()`.
```rs
let snapshot_taken = 
        self
            .snapshot_task
            .as_ref()
            .map(|t| t.is_finished())
            .unwrap_or_default();
```
to early return, we could check the inverse condition (i.e. `!snapshot_taken`)